src/MatrixClientPeg.js

```diff
@@ -48,31 +50,40 @@ class MatrixClientPeg {
         this.matrixClient = null;
     }
 
+    /**
+     * Replace this MatrixClientPeg's client with a client instance that has
+     * Home Server / Identity Server URLs but no credentials
+     */
     replaceUsingUrls(hs_url, is_url) {
-        this.replaceClient(hs_url, is_url);
+        this.replaceUsingAccessToken(hs_url, is_url);
```

Comment:
FWIW, it felt weird to me to have replaceUsingUrls call replaceUsingAccessToken with no access_token, which was why I created replaceClient. But I don't feel strongly.